addons/nightvision/functions/fnc_scotopicEffects.sqf

```diff
@@ -0,0 +1,48 @@
+#include "script_component.hpp"
+/*
+ * Author: PabstMirror
+ * Adjusts color and luminosity when in dark conditions. 
+ * Simulates rod and cones vision by decreasing color intensity when dark.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_nightvision_fnc_scotopicEffects
+ *
+ * Public: No
+ */
+private _unit = ACE_player;
+
+if (EGVAR(common,OldIsCamera) || {!alive _unit} || {currentVisionMode _unit != 0}) exitWith {
+    GVAR(scotopicCC) ppEffectEnable false;
+    [FUNC(scotopicEffects), [], 0.1] call CBA_fnc_waitAndExecute;
+};
+
+getLighting params ["", "_ambientLightBrightness"];
+(getLightingAt _unit) params ["", "", "", "_dynamiclightBrightness"];
```

Comment:
getLightingAt returns as index 1 _ambientLightBrightness so getLighting is not required.